lang/java/reef-utils/src/main/java/org/apache/reef/util/ComplexCondition.java

```diff
@@ -30,13 +30,13 @@
   private final Condition conditionVar = lockVar.newCondition();
   private final long timeoutPeriod;
   private final TimeUnit timeoutUnits;
-  private static final long DEFAULT_TIMEOUT = 10;
+  private boolean isSignal = false;
```

Comment:
I guess `isSignal` should be either `volatile` or `AtomicBoolean`

Reply:
I used volitile. The access is already protected by locks.

Reply:
I do not think this is strictly necessary as lock operations cause the necessary reads, writes and flushes to occur. From the java vm manual.

8.6 Rules About the Interaction of Locks and Variables
Let T be any thread, let V be any variable, and let L be any lock. There are certain constraints on the operations performed by T with respect to V and L:
Between an assign operation by T on V and a subsequent unlock operation by T on L, a store operation by T on V must intervene; moreover, the write operation corresponding to that store must precede the unlock operation, as seen by main memory. (Less formally: if a thread is to perform an unlock operation on any lock, it must first copy all assigned values in its working memory back out to main memory.)
Between a lock operation by T on L and a subsequent use or store operation by T on a variable V, an assign or load operation on V must intervene; moreover, if it is a load operation, then the read operation corresponding to that load must follow the lock operation, as seen by main memory. (Less formally: a lock operation behaves as if it flushes all variables from the thread's working memory, after which the thread must either assign them itself or load copies anew from main memory.)